sky/skylet/providers/lambda_labs/lambda_utils.py

```diff
@@ -0,0 +1,160 @@
+    try:
+        resp_json = response.json()
+        code = resp_json['error']['code']
+        message = resp_json['error']['message']
+    except (KeyError, json.decoder.JSONDecodeError):
+        raise LambdaLabsError(f'Unexpected error. Status code: {status_code}')
+    raise LambdaLabsError(f'{code}: {message}')
+
+
+class LambdaLabsClient:
+    """Wrapper functions for Lambda Labs API."""
+
+    def __init__(self):
+        self.credentials = os.path.expanduser(CREDENTIALS_PATH)
+        assert os.path.exists(self.credentials), 'Credentials not found'
+        with open(self.credentials, 'r') as f:
+            lines = [line.strip() for line in f.readlines() if ' = ' in line]
+            self._credentials = {
+                line.split(' = ')[0]: line.split(' = ')[1]
+                for line in lines
+            }
+        self.api_key = self._credentials['api_key']
+        self.ssh_key_name = self._credentials.get('ssh_key_name', None)
+        self.headers = {'Authorization': f'Bearer {self.api_key}'}
+
+    def up(self,
+           instance_type='gpu_1x_a100_sxm4',
+           region='us-east-1',
+           quantity=1,
+           name=''):
+        """Start a new instance."""
+        assert self.ssh_key_name is not None
+
+        # Optimization:
+        # Most API requests are rate limited at ~1 request every second but
+        # launch requests are rate limited at ~1 request every 10 seconds.
+        # So don't use launch requests to check availability.
+        # See https://docs.lambdalabs.com/cloud/rate-limiting/ for more.
+        available_regions = self.ls_catalog()['data'][instance_type]\
+                ['regions_with_capacity_available']
+        available_regions = [reg['name'] for reg in available_regions]
+        if region not in available_regions:
+            if len(available_regions) > 0:
+                aval_reg = ' '.join(available_regions)
+            else:
+                aval_reg = 'None'
+            raise LambdaLabsError(('instance-operations/launch/'
+                                   'insufficient-capacity: Not enough '
+                                   'capacity to fulfill launch request. '
+                                   'Regions with capacity available: '
+                                   f'{aval_reg}'))
+
+        # Try to launch instance
+        data = json.dumps({
+                    'region_name': region,
+                    'instance_type_name': instance_type,
+                    'ssh_key_names': [
+                        self.ssh_key_name
+                    ],
+                    'quantity': quantity,
+                    'name': name
+                })
+        response = requests.post(f'{API_ENDPOINT}/instance-operations/launch',
+                                 data=data,
+                                 headers=self.headers)
+        raise_lambda_error(response)
+        return response.json()
+
+    def rm(self, *instance_ids):
```

Comment:
I initially had command line names because I was using the `fire` library to autogenerate a CLI, but you can use names like `terminate` or `list_instances` if that seems more in style.